### PR TITLE
WakeOnLan: Add network device discovery 

### PR DESCRIPTION
**Pull request by t-mon**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

File: wakeonlan/integrationpluginwakeonlan.cpp [hunk 1]
```diff
@@ -41,19 +43,64 @@ IntegrationPluginWakeOnLan::IntegrationPluginWakeOnLan()
 {
 }
 
+void IntegrationPluginWakeOnLan::discoverThings(ThingDiscoveryInfo *info)
+{
+    if (!hardwareManager()->networkDeviceDiscovery()->available()) {
+        qCWarning(dcWakeOnLan()) << "Failed to discover network devices. The network device discovery is not available.";
+        info->finish(Thing::ThingErrorHardwareNotAvailable, QT_TR_NOOP("The discovery is not available."));
```
Comment (mzanetti) [inline wakeonlan/integrationpluginwakeonlan.cpp:50]: This isn't a very user friendly error message... IMO something like "Unable to discovery devices in your network." would be better for an end user. This one is fine for a warning message in the logs.

File: wakeonlan/integrationpluginwakeonlan.cpp [hunk 2]
```diff
@@ -41,19 +43,64 @@ IntegrationPluginWakeOnLan::IntegrationPluginWakeOnLan()
...
+        return;
+    }
+
+    qCDebug(dcWakeOnLan()) << "Start discovering network devices...";
```
Comment (mzanetti) [inline wakeonlan/integrationpluginwakeonlan.cpp:54]: "Starting network discovery..."

File: wakeonlan/integrationpluginwakeonlan.cpp [hunk 3]
```diff
@@ -41,19 +43,64 @@ IntegrationPluginWakeOnLan::IntegrationPluginWakeOnLan()
...
+    NetworkDeviceDiscoveryReply *discoveryReply = hardwareManager()->networkDeviceDiscovery()->discover();
+    connect(discoveryReply, &NetworkDeviceDiscoveryReply::finished, this, [=](){
+        ThingDescriptors descriptors;
+        qCDebug(dcWakeOnLan()) << "Discovery finished. Found" << discoveryReply->networkDeviceInfos().count() << "devices";
+        foreach (const NetworkDeviceInfo &networkDeviceInfo, discoveryReply->networkDeviceInfos()) {
+            // We need the mac address...
+            if (networkDeviceInfo.macAddress().isEmpty())
+                continue;
+
+            // Filter out already added network devices, rediscovery is in this case no option
```
Comment (mzanetti) [inline wakeonlan/integrationpluginwakeonlan.cpp:64]: Hmm... Why preventing the rediscovery? I agree it doesn't make sense, but the API allows to do it so IMO it should return something sensible.

Comment (mzanetti) [reply wakeonlan/integrationpluginwakeonlan.cpp:64]: Would be enough to change the code like this I guess:

```
Thing *existingThing = myThings().findByParam(Param(wolThingMacParamTypeId, networkDeviceInfo.macAddress());
if (existingThing) {
    descriptor.setThingId(existingThing->id());
}
```

**Review by mzanetti — CHANGES_REQUESTED** [review]
Minor nitpicks. Mostly fine.

File: wakeonlan/integrationpluginwakeonlan.cpp (hunk 3 above)
Comment (t-mon) [reply wakeonlan/integrationpluginwakeonlan.cpp:64]: The idea was: if you reconfigure this thing, you probably wanna change the mac address, if not, add a new one, yes, but a rediscovery and showing only the one you already added makes no sense, that's why I didn't wanted to set the existing id...

Comment (mzanetti) [reply wakeonlan/integrationpluginwakeonlan.cpp:64]: I see it mostly like a "Hmm... something didn't seem to work, lemme try reconfiguring it". And then the user goes to reconfigure and reconfigure doesn't work either...

**Pull request merged**